src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -62,6 +64,24 @@ const {
   Deletion,
 } = ReactTypeOfSideEffect;
 
+function transferRef(fiber: Fiber, element: ReactElement<any>) {
+  if (typeof element.ref === 'string' && element._owner) {
+    // Should we check if this string refs is equal to the previous string ref
+    // and reuse it?
+    const ownerFiber : ?Fiber = (element._owner : any);
+    if (ownerFiber) {
+      const inst = ownerFiber.stateNode;
+      const ref = function(value) {
+        const refs = inst.refs === emptyObject ? (inst.refs = {}) : inst.refs;
+        refs[element.ref] = value;
```

Comment:
We should store the ref string in a separate variable so that we don't capture the entire element and all its props in the closure.